src/libcore/macros.rs

```diff
@@ -32,8 +32,19 @@ macro_rules! panic {
 
 /// Ensure that a boolean expression is `true` at runtime.
 ///
-/// This will invoke the `panic!` macro if the provided expression cannot be
-/// evaluated to `true` at runtime.
+/// This will ensure the termination of the program if the provided expression
+/// cannot be evaluated to `true` at runtime by means of an unrecoverable error
+/// (not necessarily a `panic!`, can also be an `abort`).
+///
+/// Assertions are always checked in both debug and release builds, and cannot
+/// be disabled.
```

Comment:
It would be nice to point to `assert_debug` directly, if you want this behavior.